ns/availability.go

```diff
@@ -0,0 +1,51 @@
+package ns
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// AvailabilityRequest The structure of the request will be made to the availability endpoint.
+type AvailabilityRequest struct {
+	Credentials            Credentials     `json:"credentials"`
```

Comment:
@VictorAvelar 

Because of the ingenious way that authentication is handled by Nausys, I don't know if there's a way around passing Credentials with every request.

I was hoping for maybe adding it in the NewAPIRequest method, but I am not sure if this is possible and if this is the accepted go way. The thought was in NewAPIRequest, we fetch the credentials from the env file, and 'push' it onto the request, but like I said, not sure if this is possible?

Will it need to be present in each request that comes in? Is it something we can add ourselves? Or is there another solution that you can suggest?